pkg/collect/k8s_metrics.go

```diff
@@ -0,0 +1,127 @@
+package collect
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/metrics/pkg/apis/custom_metrics"
+)
+
+const (
+	namespaceSingular = "namespace"
+	namespacePlural   = "namespaces"
+	urlBase           = "/apis/custom.metrics.k8s.io/v1beta1"
+	metricsErrorFile  = "metrics/errors.json"
+)
+
+type CollectMetrics struct {
+	Collector    *troubleshootv1beta2.CustomMetrics
+	BundlePath   string
+	ClientConfig *rest.Config
+	Client       kubernetes.Interface
+	Context      context.Context
+	RBACErrors
+}
+
+func (c *CollectMetrics) Title() string {
+	return getCollectorName(c)
+}
+
+func (c *CollectMetrics) IsExcluded() (bool, error) {
+	return isExcluded(c.Collector.Exclude)
+}
+
+func (c *CollectMetrics) Collect(progressChan chan<- interface{}) (CollectorResult, error) {
+	output := NewResult()
+	resultLists := make(map[string][]custom_metrics.MetricValue)
+	errorsList := make([]string, 0)
+	for _, metricRequest := range c.Collector.MetricRequests {
```

Comment:
Some logging here (using `klog.V(2).Info...`) would come in handy here. When we run `./support-bundle spec.yaml -v2` we usually print verbose output of what collectors/analysers/redactors are doing. `-v1` is verbose output of what the "framework" is doing i.e support bundle, preflight then `-v0` is for high level events.

PS: Instrumenting the library with more logs is work in progress. You can take a look at how https://github.com/replicatedhq/troubleshoot/blob/main/pkg/collect/host_subnetavailable.go collector presents its progress as logs.

Reply:
sounds good, I added some logs to track progress of the requests and processing.